pkg/provisioner/ironic/ironic.go

```diff
@@ -70,6 +88,29 @@ func init() {
 		fmt.Fprintf(os.Stderr, "Cannot start: No IRONIC_INSPECTOR_ENDPOINT variable set")
 		os.Exit(1)
 	}
+
+	if authStrategy == "http_basic" {
+		ironicUser = os.Getenv("IRONIC_HTTP_BASIC_USERNAME")
+		ironicPassword = os.Getenv("IRONIC_HTTP_BASIC_PASSWORD")
+		inspectorUser = os.Getenv("INSPECTOR_HTTP_BASIC_USERNAME")
+		inspectorPassword = os.Getenv("INSPECTOR_HTTP_BASIC_PASSWORD")
```

Comment:
It seems [safer](https://twitter.com/IanColdwater/status/1079408591583526913) to me to mount the secrets as a volume rather than pass them as environment variables.